### PR TITLE
Create component metadata for secretstores.hashicorp.vault

### PR DESCRIPTION
**Pull request by MregXN**
# Description

Create component metadata for secretstores.hashicorp.vault

## Issue reference

Please reference the issue: #2585.

### CODE REVIEW

File: secretstores/hashicorp/vault/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    example: "myprefix"
+    type: string
+  - name: vaultKVUsePrefix
+    required: false
+    description: If false, vaultKVPrefix is forced to be empty. If the value is not given or set to true, vaultKVPrefix is used when accessing the vault. Setting it to false is needed to be able to use the BulkGetSecret method of the store.
+    example: "true"
+    type: bool
+  - name: enginePath
+    required: false
+    description: The engine path in vault. Defaults to secret
```
[the remark this answers is not in the document]
Comment (MregXN) [reply secretstores/hashicorp/vault/metadata.yaml]: Updated in latest commit, thanks.

File: secretstores/hashicorp/vault/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    url: "https://docs.dapr.io/reference/components-reference/supported-secret-stores/hashicorp-vault/"
+metadata:
+  - name: vaultAddr
+    required: false
+    description: The address of the Vault server. Defaults to "https://127.0.0.1:8200"
+    example: "https://127.0.0.1:8200"
+    type: string
+  - name: caPem
+    required: false
+    description: The inlined contents of the CA certificate to use, in PEM format. If defined, takes precedence over 'caPath' and 'caCert'.
```
[the remark this answers is not in the document]
Comment (MregXN) [reply secretstores/hashicorp/vault/metadata.yaml]: Updated in latest commit, thanks.

File: secretstores/hashicorp/vault/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    example: "-----BEGIN PUBLIC KEY-----\n...Base64 encoding of the DER encoded certificate...\n-----END PUBLIC KEY-----"
```
[the remark this answers is not in the document]
Comment (MregXN) [reply secretstores/hashicorp/vault/metadata.yaml]: Updated in latest commit, thanks.

File: secretstores/hashicorp/vault/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    example: "path/to/file"
+    type: string
+  - name: vaultToken
+    required: true
+    description: Token for authentication within Vault.
+    example: "tokenValue"
+    type: string
+  - name: vaultKVPrefix
+    required: false
+    description: The prefix in vault. Defaults to dapr
```
[the remark this answers is not in the document]
Comment (MregXN) [reply secretstores/hashicorp/vault/metadata.yaml]: Updated in latest commit, thanks.

File: secretstores/hashicorp/vault/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+version: v1
+status: stable
+title: "Hashicorp Vault"
+urls:
+  - title: Reference
+    url: "https://docs.dapr.io/reference/components-reference/supported-secret-stores/hashicorp-vault/"
+metadata:
+  - name: vaultAddr
+    required: false
+    description: The address of the Vault server. Defaults to "https://127.0.0.1:8200"
```
[the remark this answers is not in the document]
Comment (MregXN) [reply secretstores/hashicorp/vault/metadata.yaml]: Updated in latest commit, thanks.

File: secretstores/hashicorp/vault/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    example: "true"
+    type: bool
+  - name: enginePath
+    required: false
+    description: The engine path in vault. Defaults to secret
+    example: "kv"
+    type: string
+  - name: vaultValueType
+    required: false
+    description: Vault value type. map means to parse the value into map[string]string, text means to use the value as a string. 'map' sets the multipleKeyValuesPerSecret behavior. text makes Vault behave as a secret store with name/value semantics. Defaults to map
```
[the remark this answers is not in the document]
Comment (MregXN) [reply secretstores/hashicorp/vault/metadata.yaml]: Updated in latest commit, thanks.

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**